src/layer/hardsigmoid.cpp

```diff
@@ -0,0 +1,54 @@
+// Tencent is pleased to support the open source community by making ncnn available.
+//
+// Copyright (C) 2017 THL A29 Limited, a Tencent company. All rights reserved.
+//
+// Licensed under the BSD 3-Clause License (the "License"); you may not use this file except
+// in compliance with the License. You may obtain a copy of the License at
+//
+// https://opensource.org/licenses/BSD-3-Clause
+//
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+#include "hardsigmoid.h"
+#include <algorithm>
+
+namespace ncnn {
+
+DEFINE_LAYER_CREATOR(HardSigmoid)
+
+HardSigmoid::HardSigmoid()
+{
+    one_blob_only = true;
+    support_inplace = true;
+}
+
+int HardSigmoid::forward_inplace(Mat& bottom_top_blob, const Option& opt) const
+{
+    int w = bottom_top_blob.w;
+    int h = bottom_top_blob.h;
+    int channels = bottom_top_blob.c;
+    int size = w * h;
+
+    #pragma omp parallel for num_threads(opt.num_threads)
+    for (int q=0; q<channels; q++)
+    {
+        float* ptr = bottom_top_blob.channel(q);
+
+        for (int i=0; i<size; i++)
+        {
+            if (ptr[i] < -2.5f)
```

Comment:
y = max(0, min(1, alpha * x + beta))
the alpha and beta value should be parameterized, and loaded from param file
https://github.com/onnx/onnx/blob/master/docs/Operators.md#HardSigmoid